python-checks/src/test/resources/checks/identityComparisonWithNewObjects.py

```diff
@@ -0,0 +1,133 @@
+def basicPositives(param):
+    param is {1: 2}  # Noncompliant {{Replace this "is" operator with "==".}}
+    #     ^^ 
+    param is not {1, 2, 3}  # Noncompliant {{Replace this "is not" operator with "!=".}}
+    #     ^^^^^^
+    param is [1, 2, 3]  # Noncompliant {{Replace this "is" operator with "==".}}
+    #     ^^
+    param is dict(a=1)  # Noncompliant {{Replace this "is" operator with "==".}}
+    #     ^^
+    mylist = []  # mylist is assigned a new object
+    param is mylist  # Noncompliant {{Replace this "is" operator with "==".}}
+    #     ^^
+
+def basicNegatives(param):
+    param == {1: 2}
+    param != {1, 2, 3}
+    param == [1, 2, 3]
+    param == dict(a=1)
+    mylist = []
+    param == mylist
+
+def comprehensions(p):
+  p is { x: x for x in range(10) } # Noncompliant {{Replace this "is" operator with "==".}}
+  # ^^
+  p is not [ x for x in range(10) ] # Noncompliant {{Replace this "is not" operator with "!=".}}
+  # ^^^^^^
+  p is { x for x in range(10) } # Noncompliant {{Replace this "is" operator with "==".}}
+  # ^^
+
+def escape_00(p):
+  d = dict(a=100)
+  d is p # Noncompliant {{Replace this "is" operator with "==".}}
+  # ^^
+
+def escape_01(p):
+  d = dict(a=100)
+  save("something", d)
+  e = load("something")
+  e is d # ok, `d` could have escaped, and actually ended up in `e`
+
+def escape_02(p):
+  d = list(1, 2, 3)
+  e = d
+  e is not d # ok, `d` really did end up in `e`.
+
+def escape_03(p):
+  x = dict(a = 1)
+  #   ^^^^^^^^^^^> 1 {{This expression creates a new object every time.}}
+  x is p # Noncompliant {{Replace this "is" operator with "==".}}
+  # ^^
+  p is x # Noncompliant {{Replace this "is" operator with "==".}}
+  # ^^
+
+def coverage():
+  a[b] is b[a]
+
+x is y # should break as soon as symbols are resolved outside of functions
```

Comment:
I'm not sure what your intention is here, but symbols are actually resolved outside functions. What we don't do in the module scope is type inference.

Here none of the symbols exist as neither x nor y is defined in the module scope. If you place this snippet in a function, it still won't raise. If you define either `x = []` or `y = []` on the line above, it will raise an issue.

I think the test as it is, is fine as it ensures we don't raise on undefined symbols, but I think the comment is a bit confusing.